.github/actions/deploy-template-operator-with-modulereleasemeta/action.yml

```diff
@@ -70,14 +79,27 @@ runs:
       shell: bash
       run: |
         ./deploy_modulereleasemeta.sh ${{ env.ModuleName }} fast:${{ env.NewerVersion }} regular:${{ env.OlderVersion }}
-    - name: Create Template Operator Module as Mandatory Module
-      working-directory: lifecycle-manager
+
+    - name: Create and apply Template Operator Module as Mandatory Module
+      working-directory: template-operator
       if: ${{ matrix.e2e-test == 'mandatory-module' ||
         matrix.e2e-test == 'mandatory-module-metrics'
         }}
       shell: bash
       run: |
-        kubectl apply -f tests/e2e/moduletemplate/mandatory_moduletemplate_template_operator_v1.yaml
+        make build-manifests
+        yq eval '(. | select(.kind == "Deployment") | .metadata.name) = "${{ env.ModuleDeploymentNameInOlderVersion }}"' -i template-operator.yaml
+        ./deploy_moduletemplate.sh ${{ env.ModuleName }} ${{ env.OlderVersionForMandatoryModule }} true true
+
+    - name: Create ModuleTemplate in a new version for Mandatory module
+      if: ${{ matrix.e2e-test == 'mandatory-module'}}
+      working-directory: template-operator
+      shell: bash
+      run: |
+        make build-manifests
+        yq eval '(. | select(.kind == "Deployment") | .metadata.name) = "${{ env.ModuleDeploymentNameInNewerVersion }}"' -i template-operator.yaml
+        ./deploy_moduletemplate.sh ${{ env.ModuleName }} ${{ env.NewerVersionForMandatoryModule }} true true false
```

Comment:
Personally I find it a bit cumbersome to lookup the arguments of such scripts. For the create test clsuters one, I also changed it to named arguments: https://github.com/kyma-project/lifecycle-manager/blob/main/scripts/tests/create_test_clusters.

What do you think? Does it make sense for you if we introduce this to all scripts, as a follow-up of course?

Reply:
yes will do so, I wasn't sure how to have them named to be honest :D

Reply:
Also took some help from copilot for this :D

Reply:
https://github.com/kyma-project/lifecycle-manager/issues/2136